kyuubi-common/src/main/scala/org/apache/kyuubi/service/authentication/LdapAuthenticationProviderImpl.scala

```diff
@@ -58,29 +59,57 @@ class LdapAuthenticationProviderImpl(conf: KyuubiConf) extends PasswdAuthenticat
     conf.get(AUTHENTICATION_LDAP_URL).foreach(env.put(Context.PROVIDER_URL, _))
 
     val domain = conf.get(AUTHENTICATION_LDAP_DOMAIN)
-    val u =
-      if (!hasDomain(user) && domain.nonEmpty) {
-        user + "@" + domain.get
-      } else {
-        user
-      }
-
-    val guidKey = conf.get(AUTHENTICATION_LDAP_GUIDKEY)
```

Comment:
seems AUTHENTICATION_LDAP_GUIDKEY is not used now.

Can we remove it? cc @pan3793

Reply:
Is it a break change?

Reply:
it is a break change, we should keep backward compatibility.

@dev-lpq 

could you deprecate AUTHENTICATION_LDAP_GUIDKEY?

If the AUTHENTICATION_LDAP_BINDDN is specified, using it first.

Otherwise, fallback to use original logical to get SECURITY_PRINCIPAL?